hid-uclogic-rdescs.c

```diff
@@ -0,0 +1,665 @@
+        0x14,               /*          Logical Minimum (0),        */
+        0xA4,               /*          Push,                       */
+        0x05, 0x01,         /*          Usage Page (Desktop),       */
+        0x55, 0xFD,         /*          Unit Exponent (-3),         */
+        0x65, 0x13,         /*          Unit (Inch),                */
+        0x34,               /*          Physical Minimum (0),       */
+        0x09, 0x30,         /*          Usage (X),                  */
+        0x46, 0x10, 0x27,   /*          Physical Maximum (10000),   */
+        0x27, 0x3F, 0x9C,
+        0x00, 0x00, /*          Logical Maximum (39999),    */
+        0x81, 0x02,         /*          Input (Variable),           */
+        0x09, 0x31,         /*          Usage (Y),                  */
+        0x46, 0x6A, 0x18,   /*          Physical Maximum (6250),    */
+        0x26, 0xA7, 0x61,   /*          Logical Maximum (24999),    */
+        0x81, 0x02,         /*          Input (Variable),           */
+        0xB4,               /*          Pop,                        */
+        0x09, 0x30,         /*          Usage (Tip Pressure),       */
+        0x26, 0xFF, 0x03,   /*          Logical Maximum (1023),     */
+        0x81, 0x02,         /*          Input (Variable),           */
+        0xC0,               /*      End Collection,                 */
+        0xC0                /*  End Collection                      */
+};
+
+/* Fixed TWHA60 report descriptor, interface 1 (frame buttons) */
+const __u8 uclogic_twha60_rdesc_fixed1[] = {
+        0x05, 0x01, /*  Usage Page (Desktop),       */
+        0x09, 0x06, /*  Usage (Keyboard),           */
+        0xA1, 0x01, /*  Collection (Application),   */
+        0x85, 0x05, /*      Report ID (5),          */
+        0x05, 0x07, /*      Usage Page (Keyboard),  */
+        0x14,       /*      Logical Minimum (0),    */
+        0x25, 0x01, /*      Logical Maximum (1),    */
+        0x75, 0x01, /*      Report Size (1),        */
+        0x95, 0x08, /*      Report Count (8),       */
+        0x81, 0x01, /*      Input (Constant),       */
+        0x95, 0x0C, /*      Report Count (12),      */
+        0x19, 0x3A, /*      Usage Minimum (KB F1),  */
+        0x29, 0x45, /*      Usage Maximum (KB F12), */
+        0x81, 0x02, /*      Input (Variable),       */
+        0x95, 0x0C, /*      Report Count (12),      */
+        0x19, 0x68, /*      Usage Minimum (KB F13), */
+        0x29, 0x73, /*      Usage Maximum (KB F24), */
+        0x81, 0x02, /*      Input (Variable),       */
+        0x95, 0x08, /*      Report Count (8),       */
+        0x81, 0x01, /*      Input (Constant),       */
+        0xC0        /*  End Collection              */
+};
+
+/* Report descriptor template placeholder IDs */
+enum uclogic_ph_id {
+    UCLOGIC_PH_ID_X_LM,
+    UCLOGIC_PH_ID_X_PM,
+    UCLOGIC_PH_ID_Y_LM,
+    UCLOGIC_PH_ID_Y_PM,
+    UCLOGIC_PH_ID_PRESSURE_LM,
+    UCLOGIC_PH_ID_NUM
+};
```

Comment:
The declaration of this enum should appear in `hid-uclogic-rdescs.h` instead (please choose a good spot for it). I'm not even sure why it works as is. Also, please note the indentation.